fil_fungible_token/src/token/mod.rs

```diff
@@ -101,20 +102,38 @@ where
         self.state = mutable_state;
         Ok(res)
     }
+
+    /// Resolves an address to an ID address, sending a message to initialise an account there if
+    /// it doesn't exist
+    ///
+    /// If the account cannot be created, this function returns an error
+    fn resolve_to_id_addr(&self, initial_holder: &Address) -> Result<ActorID> {
+        let holder_id = match self.msg.resolve_id(initial_holder) {
+            Ok(addr) => addr,
+            Err(_) => self.msg.initialise_account(initial_holder)?,
+        };
+        Ok(holder_id)
+    }
+
+    /// Attempts to resolve an address to an ID address, returning an error if it did not exist
+    fn get_id_addr(&self, initial_holder: &Address) -> Result<ActorID> {
+        Ok(self.msg.resolve_id(initial_holder)?)
+    }
 }
 
-impl<BS, MC> Token<BS, MC>
+impl<BS, MSG> Token<BS, MSG>
 where
     BS: IpldStore + Clone,
-    MC: MethodCaller,
+    MSG: Messaging,
 {
     /// Mints the specified value of tokens into an account
     ///
-    /// The mint amount must be non-negative or the method returns an error
+    /// The minter is implicitly defined as the caller of the actor.
+    /// The mint amount must be non-negative or the method returns an error.
     pub fn mint(
         &mut self,
         minter: ActorID,
```

Comment:
Even though we know it will be an ID address, take `Address` here because that's what the runtime gives us. Put the `.id().map_err()` noise inside the token library rather than require it at call sites.